### PR TITLE
RFR: Admin page for changing active RO year

### PR DESCRIPTION
**Pull request by XuanLi-Leong**
## Status: 

:rocket: Ready

## Description
Change RO year

Related issues: Does some admin stuff whatever issue it is

## Screenshots

image

image

image

image

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -68,14 +69,26 @@ class App extends Component {
...
-            <Route path="/profile/badges" component={BadgesPage} />
             <Route path="/about" component={About} />
+            {/*Account pages*/}
+            <Route
+              path="/account"
+              component={withAuthenticatedProtection(AccountManagePage)}
+            />
+            <Route
+              path="/profile/badges"
+              component={withAuthenticatedProtection(BadgesPage)}
```
Comment (Davidw1339) [inline frontend/src/App.js]: yay for protection! :shield:

File: frontend/src/components/AdminNavigator.js [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
+          <ul>
+            <li>
+              <Link to="/admin/settings">Manage</Link>
+            </li>
+          </ul>
+          <b>Books</b>
           <ul>
             <li>
-              <a href="/admin/book/csv_upload">Upload CSV</a>
+              <Link to="/admin/book/csv_upload">Upload CSV</Link>
```
Comment (Davidw1339) [inline frontend/src/components/AdminNavigator.js]: nice change

File: frontend/src/components/AdminNavigator.js [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
-          Quizzes
+          <b>Parent Advice</b>
           <ul>
             <li>
-              <a href="/admin/quiz/add">Add</a>
+              <Link to="/admin/parents/add">Add</Link>
             </li>
             <li>
-              <a href="/admin/quiz/edit">Edit</a>
+              <Link to="/admin/parents/edit">Edit</Link>
```
Comment (Davidw1339) [inline frontend/src/components/AdminNavigator.js]: thanks for setting up all the routing! :twisted_rightwards_arrows:

File: frontend/src/components/admin/AdminPublishModal.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          className={this.props.className}
+        >
+          <ModalHeader toggle={this.toggle}>Confirm</ModalHeader>
+          <ModalBody>
+            Are you sure you want to publish <i>{this.props.objectName}</i>?
+          </ModalBody>
+          <ModalFooter>
+            <Button color="danger" onClick={this.props.handleConfirm}>
+              Publish
+            </Button>{' '}
```
Comment (Davidw1339) [inline frontend/src/components/admin/AdminPublishModal.js]: oh is the `{' '}` to add spacing?

File: frontend/src/utils/api.js [hunk 5]
```diff
@@ -63,6 +63,34 @@ function getROCurrentYear() {
...
+  //   method: 'POST',
+  //   body: JSON.stringify({ year: readingOlympicsYear})
+  // }).then(res => res.json());
+}
+
+function publishAllBooksInYear(readingOlympicsYear) {
+  return {
+    success: true,
+    result: {
+      // idk does it return anything?
```
Comment (Davidw1339) [inline frontend/src/utils/api.js]: just returns success i believe nothing of interest

File: frontend/src/components/admin/AdminPublishModal.js (hunk 4 above)
Comment (XuanLi-Leong) [reply frontend/src/components/admin/AdminPublishModal.js]: don't need this

File: frontend/src/utils/api.js (hunk 5 above)
Comment (XuanLi-Leong) [reply frontend/src/utils/api.js]: oops ok

**Pull request merged**